tests/Kernel/CorporateBlocks/CorporateBlocksFooterTest.php

```diff
@@ -5,316 +5,182 @@
 namespace Drupal\Tests\oe_theme\Kernel\CorporateBlocks;
 
 use Symfony\Component\DomCrawler\Crawler;
+use Drupal\Component\Render\FormattableMarkup;
 
 /**
  * Test footer block rendering.
  */
 class CorporateBlocksFooterTest extends CorporateBlocksTestBase {
 
   /**
-   * Test European Commission footer block rendering.
+   * Test European Commission footer core block rendering.
    */
-  public function testEcFooterBlockRendering(): void {
-    // Override config "oe_corporate_blocks.ec_data.footer" with test.
-    /* @var $config_obj \Drupal\Core\Config\Config */
-    $config_obj = \Drupal::service('config.factory')->getEditable('oe_corporate_blocks.ec_data.footer');
-    $test_data = $this->getFixtureContent('ec_footer.yml');
-    $config_obj->setData($test_data);
-    $config_obj->save();
-
-    // Setup and render footer block.
-    $config = [
-      'id' => 'oe_corporate_blocks_ec_footer',
-      'label' => 'OpenEuropa footer block',
-      'provider' => 'oe_corporate_blocks',
-      'label_display' => '0',
-    ];
-
-    $render = $this->buildBlock('oe_corporate_blocks_ec_footer', $config);
-
-    $html = (string) $this->container->get('renderer')->renderRoot($render);
+  public function testEcFooterCoreBlockRendering(): void {
+    $test_data = [];
+    $html = $this->renderCorporateBlocksFooter('ec', $test_data);
```

Comment:
This method render the block and fetches the test data, isn't it more clear for one method to do one thing only? Let's simply:

```php
    $test_data = $this->getFixtureContent("{$type}_footer.yml");
    $html = $this->renderCorporateBlocksFooter($type, $test_data);
```